rclcpp/src/rclcpp/timer.cpp

```diff
@@ -138,3 +143,72 @@ TimerBase::exchange_in_use_by_wait_set_state(bool in_use_state)
 {
   return in_use_by_wait_set_.exchange(in_use_state);
 }
+
+void
+TimerBase::set_on_reset_callback(std::function<void()> callback)
+{
+  if (!callback) {
+    throw std::invalid_argument(
+            "The callback passed to set_on_reset_callback "
+            "is not callable.");
```

Comment:
instead of throwing exception here, what about calling `clear_on_reset_callback`?

Reply:
This is an user set callback, how would the user be notified if something went wrong and the callback is not callable?